source/extensions/filters/network/dubbo_proxy/dubbo_hessian2_serializer_impl.cc

```diff
@@ -0,0 +1,120 @@
+#include "extensions/filters/network/dubbo_proxy/dubbo_hessian2_serializer_impl.h"
+
+#include "envoy/common/exception.h"
+
+#include "common/common/assert.h"
+#include "common/common/macros.h"
+
+#include "extensions/filters/network/dubbo_proxy/hessian_utils.h"
+#include "extensions/filters/network/dubbo_proxy/message_impl.h"
+#include "extensions/filters/network/dubbo_proxy/serializer.h"
+#include "extensions/filters/network/dubbo_proxy/serializer_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace DubboProxy {
+
+std::pair<RpcInvocationSharedPtr, bool>
+DubboHessian2SerializerImpl::deserializeRpcInvocation(Buffer::Instance& buffer,
+                                                      ContextSharedPtr context) {
+  size_t total_size = 0, size;
+  // TODO(zyfjeff): Add format checker
+  std::string dubbo_version = HessianUtils::peekString(buffer, &size);
+  total_size = total_size + size;
+  std::string service_name = HessianUtils::peekString(buffer, &size, total_size);
+  total_size = total_size + size;
+  std::string service_version = HessianUtils::peekString(buffer, &size, total_size);
+  total_size = total_size + size;
+  std::string method_name = HessianUtils::peekString(buffer, &size, total_size);
+  total_size = total_size + size;
+
+  ContextImpl* context_impl = static_cast<ContextImpl*>(context.get());
```

Comment:
`ASSERT(context)?`  Why do you want to take raw pointer here?

Reply:
sorry，there is no need to convert, this part is left out when modify the code.